### PR TITLE
Unused sss

### PR DESCRIPTION
**Pull request by JasonPaulGithub**
### Summary generated by Reviewpad on 18 Jan 24 15:13 UTC
This pull request contains two patches.

Patch 1/2:
- Adds documentation updates for the wallet module.
- Updates the documentation for the methods in the `WalletClient` struct.
- Adds new examples and clarifies the functionality of the methods.

Patch 2/2:
- Makes a feature change by removing the `send_signed_spends` method from the public API.
- Adds a comment indicating that the method is currently unused.

### CODE REVIEW

File: sn_client/src/wallet.rs [hunk 1]
```diff
@@ -201,8 +279,10 @@ impl WalletClient {
     }
 
     /// Send signed spends to another wallet.
-    /// Can optionally verify the store has been successful (this will attempt to GET the Spend from the network)
-    pub async fn send_signed_spends(
```
Comment (JasonPaulGithub) [inline sn_client/src/wallet.rs:205]: The change is here. The rest of the code comes from a branch that's waiting to merge.

File: sn_client/src/wallet.rs [hunk 2]
```diff
@@ -127,7 +126,33 @@ impl WalletClient {
         self.wallet.unconfirmed_spend_requests()
     }
 
-    /// Get the payment for a given network address
+    /// Get the payment detail from a given address within the network.
```
Comment (RolandSherwin) [inline sn_client/src/wallet.rs]: I think this function returns the cached `Payment` for the provided `NetworkAddress`. I.e., we pay for addresses before we can store anything at that location, and this retrieves the payment that the client has made.

File: sn_client/src/wallet.rs [hunk 3]
```diff
@@ -156,12 +207,39 @@ impl WalletClient {
...
     /// Remove CashNote from available_cash_notes
+    //TODO: Unused
     pub fn mark_note_as_spent(&mut self, cash_note_key: UniquePubkey) {
         self.wallet.mark_note_as_spent(cash_note_key);
     }
 
-    /// Send tokens to another wallet.
-    /// Can optionally verify the store has been successful (this will attempt to GET the Spend from the network)
+    /// Send tokens to another wallet. Can also verify the store has been successful.
+    /// This will attempt to GET the Spend from the network.
```
Comment (RolandSherwin) [inline sn_client/src/wallet.rs]: The verification is carried out by performing a GET. I think this applies to all the other similar functions with the `verify` option.
```suggestion
    /// Send tokens to another wallet. Can also verify the store has been successful, by
    /// attempting to GET the Spend from the network.
```

File: sn_client/src/wallet.rs [hunk 4]
```diff
@@ -267,7 +347,8 @@ impl WalletClient {
     ///   `skipped_chunks` is the list of content already exists in network and no need to upload
     ///
     /// Note storage cost is _per record_, and it's zero if not required for this operation.
-    /// This can optionally verify the store has been successful (this will attempt to GET the cash_note from the network)
+    /// This can optionally verify the store has been successful.
+    /// This will attempt to GET the cash_note from the network
     pub async fn pay_for_storage(
         &mut self,
         content_addrs: impl Iterator<Item = NetworkAddress>,
```
Comment (RolandSherwin) [inline sn_client/src/wallet.rs]: here `verify_store = true`, so I think we can mention that the verification is mandatory here.

**Review by maqi — CHANGES_REQUESTED** [review]
The failing CIs shall not be related to the PR, having a re-run shall be enough.

However, I'm bit confused about what this PR is related to the PR 1208 & PR 1209,
as the commit msgs looks quite similiar to each other.

Also, better remove the `merge` commit in the PR.

File: sn_client/src/wallet.rs (hunk 4 above)
Comment (JasonPaulGithub) [reply sn_client/src/wallet.rs:354]: I have changed the line to this:
    /// * `verify_store` - A boolean to verify store. Set this to true for mandatory verification.
which helps to the use to understand that the verification is mandatory if set to true. 
Hope this helps :)

**Review by maqi — APPROVED** [review]

**Pull request merged**